enclave/src/c_api/c_api.cc

```diff
@@ -509,12 +511,21 @@ int XGDMatrixCreateFromEncryptedFile(const char *fname,
     }
 #ifdef __ENCLAVE__ // pass decryption key
     // FIXME consistently use uint8_t* for key bytes
-    char key[CIPHER_KEY_SIZE];
-    EnclaveContext::getInstance().get_client_key((uint8_t*) key, username);
-    //EnclaveContext::getInstance().get_client_key(fname, (uint8_t*) key);
-    *out = new std::shared_ptr<DMatrix>(DMatrix::Load(fname, silent != 0, load_row_split, true, key));
+    char* keys[num_files];
+    std::vector<const std::string> fnames_vector;
+    for (xgboost::bst_ulong i = 0; i < num_files; ++i) {
+        char key[CIPHER_KEY_SIZE];
+        EnclaveContext::getInstance().get_client_key((uint8_t*) key, usernames[i]);
+        keys[i] = (char*) malloc(sizeof(char) * CIPHER_KEY_SIZE);
+        memcpy(keys[i], key, CIPHER_KEY_SIZE);
+        fnames_vector.push_back(std::string(fnames[i]));
+    }
+    *out = new std::shared_ptr<DMatrix>(DMatrix::LoadMultiple(fnames_vector, num_files, silent != 0, load_row_split, true, keys));
+    for (int i = 0; i < num_files; ++i) {
+        free(keys[i]);
+    }
 #else
-    *out = new std::shared_ptr<DMatrix>(DMatrix::Load(fname, silent != 0, load_row_split));
+    *out = new std::shared_ptr<DMatrix>(DMatrix::Load(fnames, silent != 0, load_row_split));
```

Comment:
This will error if ever run. Should I just delete this and remove the macros in this function?

Reply:
Yeah, delete the macro.

Reply:
Done